pvlib/iotools/meteonorm.py

```diff
@@ -0,0 +1,348 @@
+        Tilt angle from horizontal plane.
+    surface_azimuth : float, default : 180
+        Orientation (azimuth angle) of the (fixed) plane. Clockwise from north
+        (north=0, east=90, south=180, west=270).
+    time_step : {'1min', '15min', '1h'}, default : '15min'
+        Frequency of the time series.
+    horizon : str or list, default : 'auto'
+        Specification of the horizon line. Can be either 'flat', 'auto', or
+        a list of 360 integer horizon elevation angles.
+    interval_index : bool, default : False
+        Index is pd.DatetimeIndex when False, and pd.IntervalIndex when True.
+        This is an experimental feature which may be removed without warning.
+    map_variables : bool, default : True
+        When true, renames columns of the Dataframe to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    url : str, optional
+        Base URL of the Meteonorm API. The ``endpoint`` parameter is
+        appended to the url. The default is
+        :const:`pvlib.iotools.meteonorm.URL`.
+
+    Raises
+    ------
+    requests.HTTPError
+        Raises an error when an incorrect request is made.
+
+    Returns
+    -------
+    data : pd.DataFrame
+        Time series data. The index corresponds to the start (left) of the
+        interval unless ``interval_index`` is set to False.
+    meta : dict
+        Metadata.
+
+    Examples
+    --------
+    >>> # Retrieve historical time series data
+    >>> df, meta = pvlib.iotools.get_meteonorm(  # doctest: +SKIP
+    ...     latitude=50, longitude=10,  # doctest: +SKIP
+    ...     start='2023-01-01', end='2025-01-01',  # doctest: +SKIP
+    ...     api_key='redacted',  # doctest: +SKIP
+    ...     endpoint='observation/training')  # doctest: +SKIP
+
+    See Also
+    --------
+    pvlib.iotools.get_meteonorm_tmy
+
+    References
+    ----------
+    .. [1] `Meteonorm
+       <https://meteonorm.com/>`_
+    .. [2] `Meteonorm API
+       <https://docs.meteonorm.com/docs/getting-started>`_
+    .. [3] `Meteonorm API reference
+       <https://docs.meteonorm.com/api>`_
+    """
+    # Relative date strings are not yet supported
+    start = pd.Timestamp(start)
+    end = pd.Timestamp(end)
+    start = start.tz_localize('UTC') if start.tzinfo is None else start
+    end = end.tz_localize('UTC') if end.tzinfo is None else end
+
+    params = {
+        'lat': latitude,
+        'lon': longitude,
+        'start': start.strftime('%Y-%m-%dT%H:%M:%SZ'),
+        'end': end.strftime('%Y-%m-%dT%H:%M:%SZ'),
+        'parameters': parameters,
+        'surface_tilt': surface_tilt,
+        'surface_azimuth': surface_azimuth,
+        'horizon': horizon,
+        'response_format': 'json',
+    }
+
+    # Allow specifying single parameters as string
+    if isinstance(parameters, str):
+        parameters = [parameters]
+
+    # convert list to string with values separated by commas
+    if not isinstance(parameters, (str, type(None))):
```

Comment:
I don't get this condition.  `parameters` cannot be a `str` at this point (due to the list-wrapping line above).  And is `None` supported?  If so, it's not documented.